### PR TITLE
CDRIVER-5708 deprecate `mongoc_<obj>_command`

### PR DESCRIPTION
**Pull request by kevinAlbs**
# Summary

Deprecate `mongoc_<obj>_command`.

Document migrating to corresponding `mongoc_<obj>_command_simple`.

Verified by [this patch build](https://spruce.mongodb.com/version/671951525bedf30007bc2545).

# Background & Motivation

See CDRIVER-5708 for motivation for deprecating.

Previous documentation [noted]((https://mongoc.org/libmongoc/1.28.0/mongoc_client_command.html)) the `mongoc_<obj>_command` functions were "superseded" by `mongoc_<obj>_command_with_opts`, `mongoc_<obj>_read_command_with_opts`, `mongoc_<obj>_write_command_with_opts`, and `mongoc_client_read_write_<obj>_with_opts`.

Documentation has been updated to recommend migrating to `mongoc_<obj>_command_simple` for simplicity. Though the underlying call to `_mongoc_cursor_run_command` [indicates a read command](https://github.com/mongodb/mongo-c-driver/blob/9e42ae331ff079fc3ea2a8bad3d69063ddb7207f/src/libmongoc/src/mongoc/mongoc-cursor.c#L890), none of the read-specific behavior appears to apply. `mongoc_<obj>_command` [prohibits retry](https://github.com/mongodb/mongo-c-driver/blob/9e42ae331ff079fc3ea2a8bad3d69063ddb7207f/src/libmongoc/tests/test-mongoc-retryable-reads.c#L173-L174). Options are not supported, so a session cannot be passed. The read concern is not inherited from the client.

### CODE REVIEW

File: src/libmongoc/examples/migrating.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+#define EXPECT(Cond)                           \
+   if (1) {                                    \
+      if (!(Cond)) {                           \
+         FAIL ("condition failed: %s", #Cond); \
+      }                                        \
+   } else                                      \
+      (void) 0
+
+int
+main (int argc, char *argv[])
```
Comment (eramongodb) [inline src/libmongoc/examples/migrating.c:40]:
```suggestion
main (void)
```
Unused parameters.

File: src/libmongoc/doc/mongoc_database_command.rst [hunk 2]
```diff
@@ -17,9 +22,9 @@ Synopsis
                            const bson_t *command,
                            const bson_t *fields,
                            const mongoc_read_prefs_t *read_prefs)
-     BSON_GNUC_WARN_UNUSED_RESULT;
+     BSON_GNUC_WARN_UNUSED_RESULT BSON_GNUC_DEPRECATED_FOR (mongoc_database_command_simple);
```
Comment (eramongodb) [inline src/libmongoc/doc/mongoc_database_command.rst:25]: I believe these attribute-like macros should be omitted from documentation (similar to how `MONGOC_EXPORT` is omitted). These control compiler output but are not meant for users to read (these documentation pages serve that purpose instead).

File: src/libmongoc/doc/mongoc_database_command.rst [hunk 3]
```diff
@@ -17,9 +22,9 @@ Synopsis
...
 
-This function is superseded by :symbol:`mongoc_database_command_with_opts()`, :symbol:`mongoc_database_read_command_with_opts()`, :symbol:`mongoc_database_write_command_with_opts()`, and :symbol:`mongoc_database_read_write_command_with_opts()`.
+This function is superseded by :symbol:`mongoc_database_command_simple()`.
```
Comment (eramongodb) [inline src/libmongoc/doc/mongoc_database_command.rst:27]: Should this be moved into the deprecation note instead?

File: src/libmongoc/examples/migrating.c [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      // mongoc_collection_command ... before ... end
+   }
+
+   {
+      // mongoc_collection_command ... after ... begin
+      bson_t reply;
+      bson_error_t error;
+      bson_t *cmd = BCON_NEW ("find", "foo", "filter", "{", "}");
+      bool ok = mongoc_collection_command_simple (coll, cmd, NULL /* read prefs */, &reply, &error);
+      if (!ok) {
```
Comment (eramongodb) [inline src/libmongoc/examples/migrating.c:186]:
```suggestion
      if (!mongoc_collection_command_simple (coll, cmd, NULL /* read prefs */, &reply, &error)) {
```
Simplification.

File: src/libmongoc/examples/migrating.c [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+      // mongoc_client_command ... before ... end
+   }
+
+   {
+      // mongoc_client_command ... after ... begin
+      bson_t reply;
+      bson_error_t error;
+      bson_t *cmd = BCON_NEW ("find", "foo", "filter", "{", "}");
+      bool ok = mongoc_client_command_simple (client, "db", cmd, NULL /* read prefs */, &reply, &error);
+      if (!ok) {
```
Comment (eramongodb) [inline src/libmongoc/examples/migrating.c:86]:
```suggestion
      if (!mongoc_client_command_simple (client, "db", cmd, NULL /* read prefs */, &reply, &error)) {
```
Simplification.

File: src/libmongoc/examples/migrating.c [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+      // mongoc_database_command ... before ... end
+   }
+
+   {
+      // mongoc_database_command ... after ... begin
+      bson_t reply;
+      bson_error_t error;
+      bson_t *cmd = BCON_NEW ("find", "foo", "filter", "{", "}");
+      bool ok = mongoc_database_command_simple (db, cmd, NULL /* read prefs */, &reply, &error);
+      if (!ok) {
```
Comment (eramongodb) [inline src/libmongoc/examples/migrating.c:136]:
```suggestion
      if (!mongoc_database_command_simple (db, cmd, NULL /* read prefs */, &reply, &error)) {
```
Simplification.

File: src/libmongoc/doc/mongoc_database_command.rst [hunk 7]
```diff
@@ -32,14 +37,37 @@ Parameters
...
+The following example uses :symbol:`mongoc_database_command`:
+
+.. literalinclude:: ../examples/migrating.c
+   :language: c
+   :dedent: 6
+   :start-after: // mongoc_database_command ... before ... begin
+   :end-before:  // mongoc_database_command ... before ... end
+   :caption: Before
+
+To migrate, use a non-deprecated alternative, like :symbol:`mongoc_database_command_simple`:
```
Comment (eramongodb) [inline src/libmongoc/doc/mongoc_database_command.rst:62]:
```suggestion
The above code block may be rewritten to use :symbol:`mongoc_database_command_simple` instead, as shown below:
```
Wording tweaks: indicate the code block below is equivalent to the code block above.

Suggestion also applies to the other doc pages.

File: src/libmongoc/examples/basic_aggregation/map-reduce-advanced.c
Comment (eramongodb) [inline src/libmongoc/examples/basic_aggregation/map-reduce-advanced.c:1]: Does this example need to be updated? Running the example locally against MongoDB 8.0.1 returns:

```
ERROR: 'inline' takes only numeric '1'
```

**Review by eramongodb — APPROVED** [review]

**Review by eramongodb — CHANGES_REQUESTED** [review]
Misclicked prior review.

File: src/libmongoc/doc/mongoc_client_command.rst [hunk 8]
```diff
@@ -3,6 +3,12 @@
 mongoc_client_command()
 =======================
 
+.. warning::
+   .. deprecated:: 1.29.0
+
+      This function is deprecated and should not be used in new code.
+      This function is superseded by :symbol:`mongoc_client_command_simple()`.
```
Comment (eramongodb) [inline src/libmongoc/doc/mongoc_client_command.rst:10]:
```suggestion
      Use :symbol:`mongoc_client_command_simple()` instead.
```
Imperative language + consistency with release notes. Also applies to the other doc pages.

**Review by eramongodb — APPROVED** [review]
Minor suggestion; otherwise, LGTM.

**Review by vector-of-bool — APPROVED** [review]
LGTM

**Pull request merged**